src/preferences.hpp

```diff
@@ -38,7 +37,16 @@ struct Preferences {
 
     UI ui;
     Piano piano;
-    pt::ptree tree;
     void load(const std::string& filename);
     void save(const std::string& filename);
 };
+
+class SettingsFileDoesntExistException : public std::exception {
+  private:
+    std::string message;
+
+  public:
+    SettingsFileDoesntExistException() : message("Settings file doesn't exist at the given path") {}
+
+    const char* what() const throw() { return message.c_str(); }
+};
```

Comment:
Personally I would keep it simple and throw `std::runtime_error("file doesn't exist")`

Reply:
I used a runtime_error and removed the custom exception, also moved the preferences related code to preferences.cpp thus removing clutter from the main_screen.cpp